AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/KMSKeystoreOperations.dfy

```diff
@@ -96,6 +96,51 @@ module {:options "/functionSyntax:4" } KMSKeystoreOperations {
     case kmsMRKeyArn(arn) => arn
   }
 
+  method GenerateRandom(
+    kmsClient: KMS.IKMSClient
+  )
+    returns (res: Result<KMS.GenerateRandomResponse, KmsError>)
+    requires kmsClient.ValidState()
+    modifies kmsClient.Modifies
+    ensures kmsClient.ValidState()
+    ensures
+      && |kmsClient.History.GenerateRandom| == |old(kmsClient.History.GenerateRandom)| + 1
+      && KMS.GenerateRandomRequest(
+           NumberOfBytes:=  Some(32),
+           CustomKeyStoreId:= None,
+           Recipient:= None
+         )
+         == Seq.Last(kmsClient.History.GenerateRandom).input
+      && old(kmsClient.History.GenerateRandom) < kmsClient.History.GenerateRandom
+      && old(kmsClient.History.Encrypt) == kmsClient.History.Encrypt
+      && old(kmsClient.History.Decrypt) == kmsClient.History.Decrypt
```

Comment:
These should not be needed.
If they are... something is wrong with Dafny/Smithy-Dafny's Types file.
```suggestion
```